Marlin/src/module/probe.cpp

```diff
@@ -51,6 +51,10 @@
 #endif
 
 float zprobe_zoffset; // Initialized by settings.load()
+#if ENABLED(DUAL_X_CARRIAGE) && ENABLED(INDEPENDENT_Z_OFFSETS)
+  float x1_zprobe_zoffset;
+  float x2_zprobe_zoffset;
```

Comment:
Menus are totally ok with using something like `zprobe_zoffset[active_extruder]` as the thing being live-edited.

Reply:
But if tool change is called while the screen is up, you wont see the value for the new extruder until the first change is made. I wanted to drop out of the menu so that you dont see a stale value.

Reply:
If there are going to be two independent offsets, it will be best to have two menu items — one for X1 and one for X2.